### PR TITLE
FEATURE: Add default editorOptions to settings

### PR DESCRIPTION
**Pull request by samuelhauser**
Add default editorOptions to settings for the new react ui [https://github.com/neos/neos-ui/issues/641](https://github.com/neos/neos-ui/issues/641)

### CODE REVIEW

File: TYPO3.Neos/Configuration/Settings.yaml [hunk 1]
```diff
@@ -312,10 +312,14 @@ TYPO3:
           'TYPO3.Neos/Inspector/Editors/ReferencesEditor':
             editorOptions:
               placeholder: 'TYPO3.Neos:Main:typeToSearch'
+              nodeTypes: ['Neos.Neos:Document']
```
Comment (kdambekalns) [inline TYPO3.Neos/Configuration/Settings.yaml]: This is the 2.3 branch, it must be `TYPO3.Neos`…

Comment (aertmann) [inline TYPO3.Neos/Configuration/Settings.yaml]: oh good catch 😮

Comment (kdambekalns) [inline TYPO3.Neos/Configuration/Settings.yaml]: Thanks to @aertmann for fixing this in https://github.com/neos/neos-development-collection/commit/3fb088cac26fe649158ae4166eb8f185975481ce

Comment (mgoldbeck) [inline TYPO3.Neos/Configuration/Settings.yaml]: thanks @aertmann

**Pull request merged**